ext/LinearSolveForwardDiff.jl

```diff
@@ -0,0 +1,80 @@
+module LinearSolveForwardDiff
+
+using LinearSolve
+isdefined(Base, :get_extension) ? 
+    (import ForwardDiff; using ForwardDiff: Dual) : 
+    (import ..ForwardDiff; using ..ForwardDiff: Dual)
+
+function _solve!(cache, alg, dAs, dbs, A, b, T; kwargs...)
+    @assert !(eltype(first(dAs)) isa Dual)
+    @assert !(eltype(first(dbs)) isa Dual)
+    @assert !(eltype(A) isa Dual)
+    @assert !(eltype(b) isa Dual)
+    reltol = cache.reltol isa Dual ? ForwardDiff.value(cache.reltol) : cache.reltol
+    abstol = cache.abstol isa Dual ? ForwardDiff.value(cache.abstol) : cache.abstol
+    u = eltype(cache.u) <: Dual ? ForwardDiff.value.(cache.u) : cache.u
+    cacheval = eltype(cache.cacheval.factors) <: Dual ? begin 
+        LinearSolve.LinearAlgebra.LU(ForwardDiff.value.(cache.cacheval.factors), cache.cacheval.ipiv, cache.cacheval.info)
+    end : cache.cacheval
+    cache2 = remake(cache; A, b, u, reltol, abstol, cacheval)
```

Comment:
Being forced to remake cache in order to solve the non-dual version. Is there some other way we can replace Dual Array with a regular array?

Reply:
I think you want to hook into `init`. In theory in `init` what you can do is un-dual the user inputs that are dual, but tag the cache in such a way that in `solve!` you end up doing two (or number of chunk size + 1) solves and reconstruct the resulting dual numbers in the output.

Reply:
Or rather, it's just one `solve!` call but in a batched form.